### ISSUE TITLE
Date/Time field error during Pods Edit on iPad 

### ISSUE THREAD

**Issue by lisaschiffli** [pods-framework/pods#6318]
### Description

When editing new or existing custom pods items (ie Concerts), on iPads (both Safari and Chrome, 2 diff iPads), all Date/Time fields are unavailable for editing and display the error "There was an error rendering the field." This occurs on both existing previously entered date/time entries as well as new entries. It also occurs when a new test date/time field is tried. This occurs on two different websites. My client uses an iPad to enter her data.

This does not occur on Windows 10 (Chrome, Firefox, Edge) or on client's Mac Book. The date/time fields can be edited as expected.

### Version

2.8.5

### Testing Instructions

1. Edit an existing custom pod from the Dashboard or Add New. Locate a date/time field.
2. Field will report error "There was an error rendering the field." next to the label.

### Possible Workaround

Use a Windows 10 or Mac Book instead of iPad to edit date/time pods fields.

**Comment by JoryHogeveen** [comment]
Hi @lisaschiffli ,

Please share your field config so we can validate/reproduce the issue!
Otherwise it’s very hard for us to debug 🙂

Cheers, Jory

**Comment by JoryHogeveen** [comment]
Probably the same issue: https://github.com/pods-framework/pods/issues/6300

**Comment by lisaschiffli** [comment]
Here is my date/time field configuration. 

[https://test.recklesssaints.com/files/concert_date_time_field_details.png](url)
[https://test.recklesssaints.com/files/concert_date_time_options.png](url)
[https://test.recklesssaints.com/files/concert_date_time_advanced.png](url)
[https://test.recklesssaints.com/files/concert_date_time_REST_API.png](url)

**Comment by JoryHogeveen** [comment]
Also related: https://wordpress.org/support/topic/date-fields-not-displaying-since-upgrade/

**Comment by sc0ttkclark** [comment]
@zrothauser This one could use an additional eye from you to verify it's not an issue with our DFV field itself.

I'll do some testing on my iPad to see what I can find out too.

**Comment by sc0ttkclark** [comment]
@zrothauser Confirmed, this is definitely happening.

Mobile chrome dev logs show this for every date/time/datetime field in the Kitchen Sink CPT. Not a single one of the fields works at the moment.

```
ERROR: SyntaxError: Invalid regular expression: invalid group specifier name
WARNING: There was an error rendering this field. SyntaxError: Invalid regular expression: invalid group specifier name [object Object]
```

**Comment by sc0ttkclark** [comment]
This happens on iPadOS (iPad) and iOS (iPhone) on Chrome or Safari, just shows “There was an error rendering the field” message

**Comment by sc0ttkclark** [comment]
From Safari web inspector via cable:

```
[Warning] There was an error rendering this field. (2) (pods-dfv.min.js, line 1)
SyntaxError: Invalid regular expression: invalid group specifier name
RegExp
Hu — pods-dfv.min.js:1:1146662
ec — pods-dfv.min.js:1:1149864
useMemo — react-dom.min.js:216:443
nc — pods-dfv.min.js:1:1151335
we — react-dom.min.js:84:295
zj — react-dom.min.js:226:499
Th — react-dom.min.js:152:226
tj — react-dom.min.js:152:155
Te — react-dom.min.js:146:154
Ja — react-dom.min.js:224:384
md — react-dom.min.js:173
(anonymous function) — react-dom.min.js:175:225
Rh — react-dom.min.js:147
nd — react-dom.min.js:175:211
init — pods-dfv.min.js:1:1286509
(anonymous function) — pods-dfv.min.js:1:1287056
Object

componentStack: "↵    in nc↵    in Gh↵    in i↵    in div↵    in div↵    in div↵    in zr↵    in wm↵    in wm↵    in WithDispatch(Component)↵    in Unknown…"

Object Prototype

__defineGetter__(propertyName, getterFunction)

__defineSetter__(propertyName, setterFunction)

__lookupGetter__(propertyName)

__lookupSetter__(propertyName)

constructor: function()

hasOwnProperty(propertyName)

isPrototypeOf(property)

propertyIsEnumerable(propertyName)

toLocaleString()

toString()

valueOf()
```

**Comment by jakestpeter** [comment]
I feel like this is broken in Safari but works in Chrome ....

> This happens on iPadOS (iPad) and iOS (iPhone) on Chrome or Safari, just shows “There was an error rendering the field” message

**Comment by MartvanderKruis** [comment]
Yep still the error. I have this on my macbook as well. Seems it works in Chrome but only happens in all things Safari.

**Comment by pdclark** [comment]
Verified on Safari Desktop version `15.2` — same console error regarding an invalid regex.

**Comment by bslickis** [comment]
Error still occurring in running 2.8.8.1. Only occurs on iOS devices, whether running Safari or Chrome. Works fine on Chrone and Explorer on Android and Desktop.

Field configurations for my site:

http://kaylees3.sg-host.com/wp-content/uploads/2022/01/Field-Details.png
http://kaylees3.sg-host.com/wp-content/uploads/2022/01/Date-Time-Options.png
http://kaylees3.sg-host.com/wp-content/uploads/2022/01/Advanced.png
http://kaylees3.sg-host.com/wp-content/uploads/2022/01/REST-API.png

**Comment by nobleclem** [comment]
So I ran into this issue today reported by a client and did some investigating.  The issue is a result of the crosswalk from PHP dateformat to moment dateformat RegExp.

The issue resides in: ui/js/dfv/src/helpers/dateFormats.js#L99

The `regexString` being created contains a **lookbehind** `(?<!\\\\)` in it and [Safari does not yet support lookbehinds](https://caniuse.com/js-regexp-lookbehind).

I am not sure if this is resolving a use case I am unaware of but for it to work in safari it would need to be removed.  A quick and simple check does confirm it but without knowing more about the reason for its existence I cannot say it will work without other use case errors popping up.

**Comment by mbjmcjohnston** [comment]
Still seeing same issue on Desktop Safari

**Comment by sc0ttkclark** [comment]
Fixed in 2.8.9
